File: src/24. Swap Nodes in Pairs.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */

/**
 * @param {ListNode} head
 * @return {ListNode}
 */

const swapPairs = (head) => {
  const dummy = new ListNode(undefined, head)
  let prev = dummy
  let cur = dummy.next
  while (cur !== null && cur.next !== null) {
    let tmp = cur.next.next
    cur.next.next = cur
    prev.next = cur.next
    cur.next = tmp
    prev = cur
    cur = cur.next
  }
  return dummy.next
}

File: src/24. Swap Nodes in Pairs.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* swapPairs(ListNode* head) {
        ListNode *dummy = new ListNode(-1, head),
            *prev = dummy,
            *curr = head;
        while (curr && curr->next) {
            ListNode* tmp = curr->next;
            prev->next = tmp;
            curr->next = tmp->next;
            tmp->next = curr;
            prev = curr;
            curr = curr->next;
        }
        return dummy->next;
    }
};